### PR TITLE
[DXP-995] Update quarkus version after its internal bug is fixed

### PR DESCRIPTION
**Pull request by stan-r-ds**

### CODE REVIEW

File: core/src/main/java/dev/streamx/cli/config/DotStreamxConfigSource.java [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 import static java.nio.file.StandardOpenOption.CREATE;
 
-import io.quarkus.runtime.configuration.ApplicationPropertiesConfigSourceLoader.InClassPath;
-import io.quarkus.runtime.configuration.ApplicationPropertiesConfigSourceLoader.InFileSystem;
```
Comment (stan-r-ds) [inline core/src/main/java/dev/streamx/cli/config/DotStreamxConfigSource.java:6]: No such classes anymore, in quarkus 3.13.1

File: core/src/main/java/dev/streamx/cli/config/DotStreamxConfigSource.java [hunk 2]
```diff
@@ -17,8 +15,8 @@ public class DotStreamxConfigSource extends PropertiesConfigSource {
 
   public static final String CONFIG_SOURCE_NAME = "DotStreamxConfigSource";
   /**
-   * Value higher priority than {@link InClassPath Classpath Properties}
-   * but lower than {@link InFileSystem $PWD/config/application.properties}
+   * Value of higher priority than Classpath Properties,
+   * but lower than $PWD/config/application.properties
```
Comment (stan-r-ds) [inline core/src/main/java/dev/streamx/cli/config/DotStreamxConfigSource.java:19]: I think we don't lose information when dropping the links

File: core/src/test/java/dev/streamx/cli/license/LicenseFetcherMalformedTest.java [hunk 3]
```diff
@@ -5,14 +5,13 @@
...
-import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.WithTestResource;
 import io.quarkus.test.junit.QuarkusTest;
 import jakarta.inject.Inject;
 import org.junit.jupiter.api.Test;
 
 @QuarkusTest
-@QuarkusTestResource(restrictToAnnotatedClass = true,
-    value = MalformedWiremockLicense.class)
+@WithTestResource(MalformedWiremockLicense.class)
```
Comment (stan-r-ds) [inline core/src/test/java/dev/streamx/cli/license/LicenseFetcherMalformedTest.java:14]: 1. `@QuarkusTestResource` become deprecated
2. `@WithTestResource` is the replacement suggested in deprecation comment
3. `restrictToAnnotatedClass = true` is now the default value in `WithTestResource`
4. The remaining field `value` is now the only required setting, so "value = " can be skipped

File: pom.xml [hunk 4]
```diff
@@ -37,7 +37,7 @@
 
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
-    <quarkus.platform.version>3.12.0</quarkus.platform.version>
+    <quarkus.platform.version>3.13.1</quarkus.platform.version>
```
Comment (stan-r-ds) [inline pom.xml:40]: 3.13.0 had a bug regarding returning exit status of command line processes

**Review by marekczajkowski — APPROVED** [review]

**Pull request merged**